test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java

```diff
@@ -67,18 +72,21 @@ public void handle(final HttpExchange exchange) throws IOException {
             assert read == -1 : "Request body should have been empty but saw [" + read + "]";
         }
         try {
+            // Request body is closed in the finally block
+            final BytesReference requestBody = Streams.readFully(Streams.noCloseStream(exchange.getRequestBody()));
```

Comment:
Since we needed the request body again I refactored things like we did for the other mocks to full drain the input first thing here as well to keep things simple.